cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -444,11 +445,12 @@ public int hashCode() {
     public PluginClassLoader load(ClassLoaderKey key) throws Exception {
       File unpackedDir = DirUtils.createTempDir(tmpDir);
       File artifact = new File(pluginDir, Artifacts.getFileName(key.artifact));
-      BundleJarUtil.unJar(Locations.toLocation(artifact), unpackedDir);
+      BundleJarUtil.unJar(Locations.toLocation(artifact), unpackedDir, name ->
+        name.equals(JarFile.MANIFEST_NAME) || name.endsWith(".jar"));
```

Comment:
I'm trying to think if we have any use cases where we're relying on some other files to be bundled with the jar and extracted. @chtyim do you know of any? 

I know for initial experiments with an SAP plugin that native libraries were bundled in the jar, but we eventually separated them out.

Reply:
Hum... technically only .jar files need to be extracted as all other files should be loaded via ClassLoader.

@tivv when the Manifest file needs to be expanded?

Reply:
I did not want to touch the code in io.cdap.cdap.common.lang.DirectoryClassLoader#DirectoryClassLoader(java.io.File, java.lang.String, java.lang.ClassLoader, java.lang.Iterable<java.lang.String>)
```
      try (
        InputStream input = new FileInputStream(new File(dir, JarFile.MANIFEST_NAME.replace('/', File.separatorChar)))
      ) {
        manifest = new Manifest(input);
      }
```
I can migrate it though.
Also I am fixing now com.google.common.collect.AbstractIterator#computeNext as it relies on all ".class" be files

Reply:
@albertshau anything that works through the ClassLoader and can take "jar:" URIs would see everything as before. If something relies on those things be files, it would need to be fixed, e.g. `io.cdap.cdap.internal.app.runtime.artifact.ArtifactInspector#getPluginClasses`

Reply:
I looked at the manifest and I don't think I want to touch it. It's in the DirectoryClassLoader that is overriden by two other classes and I should be extra careful to pick up  a correct manifest (and not the one from the different jar), so I'd  better not touch it.